templates/gsl.install.sh

```diff
@@ -272,202 +398,134 @@ build_from_tarball_boost()
         "-sICU_LINK=${ICU_LIBS[@]}" \\
         "-sZLIB_LIBPATH=$PREFIX/lib" \\
         "-sZLIB_INCLUDE=$PREFIX/include" \\
-        "-j $JOBS" \\
+        "-j $PARALLEL" \\
         "-d0" \\
         "-q" \\
         "--reconfigure" \\
         "--prefix=$PREFIX" \\
-        "$@"
+        "${OPTIONS[@]}"
 
     pop_directory
     pop_directory
 }
 
-.endmacro # define_build_from_tarball_boost
+.   define_boost_build_configuration_helpers()
+.endmacro #define_build_from_tarball_boost
 .
-.macro define_build_from_github()
-# Standard build from github.
-build_from_github()
+.macro define_build_from_tarball_zmq(install)
+.   define my.install = define_build_from_tarball_zmq.install
+build_from_tarball_zmq()
 {
-    push_directory "$BUILD_DIR"
+.   zmq_url = get_zmq_url(my.install)
+.   zmq_file = get_zmq_file(my.install)
+    local URL="$(zmq_url)"
+    local ARCHIVE="$(zmq_file)"
+    local COMPRESSION="gzip"    
+
+    change_dir_for_build_and_download_and_extract "$ARCHIVE" "$URL" "$COMPRESSION"   
+    configure_and_make_and_install   
 
+    pop_directory
+    pop_directory
+}
+
+.endmacro #define_build_from_tarball_zmq
+.
+.macro define_build_from_tarball_mbedlts(install)
+.   define my.install = define_build_from_tarball_mbedlts.install
+build_from_tarball_mbedlts()
```

Comment:
Code generation is meant to avoid this repetition.  Note how the zmq function above differs not at all functionally from the mbedtls and yet we have a new function per non-standard dependency as opposed to new function per procedure.